File: akka-benchmark/src/main/java/io/sarl/akka/eventbus/lookup/SubscriberActor.java
package io.sarl.akka.eventbus.lookup;

import akka.actor.AbstractActor;
import akka.actor.Props;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class SubscriberActor extends AbstractActor {
    static public Props props() {
        return Props.create(SubscriberActor.class, SubscriberActor::new);
    }

    private LoggingAdapter log = Logging.getLogger(getContext().getSystem(), this);

    public SubscriberActor() {
        //
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(Message.class, message -> {
                    log.info(message.getContent());
                })
                .build();
    }
}

File: akka-benchmark/src/main/kotlin/io/sarl/akka/eventbus/subchannel/SubscriberActor.kt
package io.sarl.akka.eventbus.subchannel

import akka.actor.AbstractActor
import akka.actor.Props
import akka.event.Logging
import akka.event.LoggingAdapter
import akka.japi.Creator

class SubscriberActor : AbstractActor() {

    private val log = Logging.getLogger(context.system, this)

    override fun createReceive(): AbstractActor.Receive {
        return receiveBuilder()
                .match(Message::class.java) { message -> log.info(message.content) }
                .build()
    }

    companion object {
        fun props(): Props {
            return Props.create<SubscriberActor>(SubscriberActor::class.java) { SubscriberActor() }
        }
    }
}//
